toolchain/check/import.cpp

```diff
@@ -152,14 +152,13 @@ static auto AddNamespace(Context& context, SemIR::TypeId namespace_type_id,
   // supersede the name conflict in order to avoid repeat diagnostics. Names are
   // poisoned optimistically by name lookup before checking for imports, so we
   // may be overwriting a poisoned entry here.
-  auto& entry = parent_scope->GetEntry(entry_id);
-  if (entry.is_poisoned) {
-    entry.is_poisoned = false;
-  } else if (!inserted) {
-    context.DiagnoseDuplicateName(namespace_id, entry.inst_id);
-    entry.access_kind = SemIR::AccessKind::Public;
+  auto& result = parent_scope->GetEntry(entry_id).result;
+  SemIR::AccessKind access_kind = result.access_kind();
+  if (!result.is_poisoned() && !inserted) {
+    context.DiagnoseDuplicateName(namespace_id, result.lookup_inst_id());
+    access_kind = SemIR::AccessKind::Public;
   }
-  entry.inst_id = namespace_id;
+  result = SemIR::ScopeLookupResult::Create(namespace_id, access_kind);
```

Comment:
Should this just always set the access kind to public when it replaces? I think the `is_poisoned` conditionality might have been incorrect. i.e., a namespace is currently always public -- and the `AddNamespace` call here has no support for non-public namespaces anyways.

i.e.:

```suggestion
  auto& result = parent_scope->GetEntry(entry_id).result;
  if (!result.is_poisoned() && !inserted) {
    context.DiagnoseDuplicateName(namespace_id, result.lookup_inst_id());
  }
  result = SemIR::ScopeLookupResult::Create(namespace_id, SemIR::AccessKind::Public);
```

Reply:
Makes sense, done.
I think this is another example of why having an explicit API helps prevent and find issues that could be caused by implicit assumptions that could end up being not correct.